dev-libs/libjwt/libjwt-1.15.3.ebuild

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools
+
+DESCRIPTION="JWT C Library"
+HOMEPAGE="https://github.com/benmcollins/libjwt"
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/benmcollins/libjwt"
+else
+	SRC_URI="https://github.com/benmcollins/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="MPL-2.0"
+SLOT="0"
+# openssl / gnutls set which ssl implementations to use (build libjwt-ossl.so / libjwt-gnutls.so)
+# IF openssl is enabled it will be the implementation used for libjwt.so
+# gnutls will only be used for libjwt.so if openssl is disabled
+IUSE="gnutls +openssl test"
+
+REQUIRED_USE="
+	|| ( gnutls openssl )
+"
```

Comment:
Hi Juippis,

Thanks for looking into this.  Please refer to the discussion in #33291 

Short version: libjwt can compile with BOTH implementations at any point in time.  So two USE flags are required.  I avoided USE=ssl because that's generally used for enabling/disabling SSL/TLS support, something which is required here in order to obtain the cryptographic tokens by way of the hash functions in these libraries.

Reply:
I think it can be simplified.
> # IF openssl is enabled it will be the implementation used for libjwt.so
> # gnutls will only be used for libjwt.so if openssl is disabled

I would only put `gnutls` use flag here: if it's disabled, choose openssl.

Reply:
I originally pushed that way.

It seems that zoneminder requires the gnutls variation, and asterisk the openssl one (although - I suspect it can use either).

The big issue is that zoneminder requires the multi-variant patch, ie, it can't just use libjwt, it has to use the one with -gnutls whatever appended.  And asterisk requires just libjwt.

Anyway, I've been back and forth and I do think this is the most sensible solution, but open for debate.  Would prefer something simpler myself, but also don't want to break zoneminder (in an overlay).